### PR TITLE
feat: [Importer] Block config custom mappings

### PR DESCRIPTION
**Pull request by arumsey**
When a import ruleset JSON is being created any block cell customizations included in the mapping objects is now peocessed as well. Currently only the **block config** flavour is supported. The result will be a `cells` object being added to each block rule that had customizations added.

Input mappings array that includes
```
[
      {
        name: 'title',
        value: 'Products',
        condition: '.product',
      },
      {
        name: 'title',
        value: 'News',
        condition: '.news',
      },
]
```

would become

```
{
      title: [
        ['.product', 'Products'],
        ['.news', 'News'],
      ],
}
```

### CODE REVIEW

File: .eslintrc.cjs [hunk 1]
```diff
@@ -22,13 +22,6 @@ module.exports = {
     'import/extensions': ['error', {
       js: 'always',
     }],
-    'comma-dangle': ["error", {
-      "arrays": "only-multiline",
-      "objects": "only-multiline",
-      "imports": "only-multiline",
-      "exports": "always",
-      "functions": "never"
-    }],
```
Comment (arumsey) [inline .eslintrc.cjs:31]: `npm run lint` now passes with this removed.

File: js/import/rules.import.js [hunk 2]
```diff
@@ -12,18 +12,18 @@
...
     start: [],
-    end: [],
   },
   blocks: [
     {
       type: 'metadata',
       insertMode: 'append',
       params: {
-        metadata: {},
+        cells: {},
```
Comment (arumsey) [inline js/import/rules.import.js:26]: A `metadata` specific param is no longer needed. All block customizations can be handled by the same `cells` property which can either be a name/value pair object or a two dimensional array of selectors.

File: js/import/rules.import.js [hunk 3]
```diff
@@ -52,7 +52,36 @@ function buildSelector(mapping, basePath) {
...
   }
-  return '';
+  return undefined;
+}
+
+/**
+ * Build a block cells object from a list of mappings.
+ * @param mappingList
+ */
+function buildBlockCellsFromMapping(mappingList = []) {
```
Comment (arumsey) [inline js/import/rules.import.js:62]: This is the new function that converts an array of mapping objects into a compatible `cells` object.

File: js/import/rules.import.js [hunk 4]
```diff
@@ -90,17 +123,29 @@ function buildTransformationRulesFromMapping(mapping) {
...
+    const existingRules = transformRules.blocks.find((b) => b.type === type);
+    const selectors = mappingList.map((m) => buildSelector(m, rootXpath)).filter((s) => s);
+    const cells = buildBlockCellsFromMapping(mappingList);
+    return {
+      ...existingRules,
+      type,
+      selectors,
+      params: { cells },
+    };
+  });
```
Comment (arumsey) [inline js/import/rules.import.js:136]: Builinding up the set of block rules for importing which is based on all the available mapping objects.

File: modules/parsers/metadata.js [hunk 5]
```diff
@@ -1,13 +1,16 @@
...
   const date = new Date(str);
-  return !Number.isNaN(date);
+  return !Number.isNaN(Number(date));
 }
 
-export default function parse(element, { document, params: { metadata = {} } }) {
+export default function parse(element, { document, params: { cells = {} } }) {
   const baseMetadata = WebImporter.Blocks.getMetadata(document) || {};
-  const customMetadata = WebImporter.Transformer.buildBlockConfig(document, metadata);
+  const customMetadata = blockParser(document, { params: { cells } });
```
Comment (arumsey) [inline modules/parsers/metadata.js:13]: Treat `metadata` block like any other block which means the generic block parser can be used.

**Review by atopper — APPROVED** [review]

**Pull request merged**